client/components/account-balances/balance-block.tsx

```diff
@@ -8,25 +8,30 @@ import * as React from 'react';
  */
 import { formatCurrency } from 'wcpay/utils/currency';
 import Loadable from 'components/loadable';
-import BalanceTooltip from './balance-tooltip';
+import {
+	TotalBalanceTooltip,
+	AvailableBalanceTooltip,
+} from './balance-tooltip';
 
 /**
  * BalanceBlockProps
  *
  * @typedef {Object} BalanceBlockProps
  *
- * @property {string} id											The balance block id. Used to link the title and amount.
- * @property {string} title											The balance title.
- * @property {string} currencyCode									Currency code of the balance block.
- * @property {React.ReactElement< typeof BalanceTooltip >} tooltip	The tooltip element.
- * @property {number} [amount]										Optional. The balance amount.
- * @property {boolean} [isLoading]									Optional. Whether the balance block is loading.
+ * @property {string} id																					The balance block id. Used to link the title and amount.
+ * @property {string} title																					The balance title.
+ * @property {string} currencyCode																			Currency code of the balance block.
+ * @property {React.ReactElement< typeof TotalBalanceTooltip | typeof AvailableBalanceTooltip >} tooltip	The tooltip element.
+ * @property {number} [amount]																				Optional. The balance amount.
+ * @property {boolean} [isLoading]																			Optional. Whether the balance block is loading.
  */
 interface BalanceBlockProps {
 	id: string;
 	title: string;
 	currencyCode: string;
-	tooltip?: React.ReactElement< typeof BalanceTooltip >;
+	tooltip?: React.ReactElement<
+		typeof TotalBalanceTooltip | typeof AvailableBalanceTooltip
+	>;
```

Comment:
Curious to hear @Jinksi 's thoughts on `BalanceBlock` component and strongly typing its props.

This feels like tight-coupling code smell to me:

- How strict do we need to be about the `tooltip` prop – e.g. can we require it to be a kind of tooltip, or a small bit of markup, or is it ok to just require that it's a renderable component (assuming that's possible in TypeScript)? 
- Is `BalanceBlock` and `BalanceBlockProps` serving us? It feels overly specific and tightly coupled to the outer block's needs. If we change how we display balances, we'll likely change `BalanceBlock` and strongly typed props. Would the code be worse if we inlined it all?

Reply:
> How strict do we need to be about the tooltip prop

Yeh I agree, `React.ReactElement` would suffice. The duplication in JSDoc types can be removed too IMO. We can add descriptive comments to the TS declarations instead.

> Would the code be worse if we inlined it all?

I'd say it would be better to inline, since it is used only twice in a single file. Might be better to refactor in a separate PR though.

Reply:
Addressed in https://github.com/Automattic/woocommerce-payments/pull/8140/commits/6f64549a8ab54035702903555bc74127bdca866e.